state/docker_resource.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/juju/errors"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"gopkg.in/mgo.v2/txn"
+	"gopkg.in/yaml.v2"
+
+	"github.com/juju/juju/core/resources"
+)
+
+// DockerResource implements DockerResourceStorage
+type DockerResource struct {
```

Comment:
This doesn't need to be exported.
Should probably be called dockerResourceStorage